dll/win32/shell32/dialogs/folder_options.cpp

```diff
@@ -1556,6 +1556,311 @@ FindSelectedItem(
+        LoadStringW(shell32_hInstance, IDS_NEWEXT_SPECIFY_EXT, szText, _countof(szText));
+        szText[_countof(szText) - 1] = 0;
+        LoadStringW(shell32_hInstance, IDS_FILE_TYPES, szTitle, _countof(szTitle));
+        szTitle[_countof(szTitle) - 1] = 0;
+        MessageBoxW(hwndDlg, szText, szTitle, MB_ICONERROR);
+        return FALSE;
+    }
+
+    ZeroMemory(&find, sizeof(find));
+    find.flags = LVFI_STRING;
+    if (pNewExt->szExt[0] == L'.')
+    {
+        find.psz = &pNewExt->szExt[1];
+    }
+    else
+    {
+        find.psz = pNewExt->szExt;
+    }
+
+    iItem = ListView_FindItem(pNewExt->hwndLV, -1, &find);
+    if (iItem >= 0)
+    {
+        // already exists
+        WCHAR szText[256], szFormat[256], szTitle[64], szFileType[64];
+
+        // get file type
+        LV_ITEM item;
+        ZeroMemory(&item, sizeof(item));
+        item.mask = LVIF_TEXT;
+        item.pszText = szFileType;
+        item.cchTextMax = _countof(szFileType);
+        item.iItem = iItem;
+        item.iSubItem = 1;
+        ListView_GetItem(pNewExt->hwndLV, &item);
+
+        // get text
+        LoadStringW(shell32_hInstance, IDS_NEWEXT_ALREADY_ASSOC, szFormat, _countof(szFormat));
+        szText[_countof(szFormat) - 1] = 0;
+        StringCchPrintf(szText, _countof(szText), szFormat, find.psz, szFileType, find.psz, szFileType);
+
+        // get title
+        LoadStringW(shell32_hInstance, IDS_NEWEXT_EXT_IN_USE, szTitle, _countof(szTitle));
+        szTitle[_countof(szTitle) - 1] = 0;
+
+        if (MessageBoxW(hwndDlg, szText, szTitle, MB_ICONWARNING | MB_YESNO) == IDNO)
+        {
+            return FALSE;
+        }
+
+        // Delete the item
+        ListView_DeleteItem(pNewExt->hwndLV, iItem);
+    }
+
+    EndDialog(hwndDlg, IDOK);
+    return TRUE;
+}
+
+// IDD_NEWEXTENSION dialog
+INT_PTR
+CALLBACK
+NewExtensionDlgProc(
+    HWND hwndDlg,
+    UINT uMsg,
+    WPARAM wParam,
+    LPARAM lParam)
+{
+    static NEWEXT_DIALOG *s_pNewExt = NULL;
+
+    switch (uMsg)
+    {
+        case WM_INITDIALOG:
+            s_pNewExt = (NEWEXT_DIALOG *)lParam;
+            NewExtDlg_OnInitDialog(hwndDlg, s_pNewExt);
+            return TRUE;
+        case WM_COMMAND:
+            switch (LOWORD(wParam))
+            {
+                case IDOK:
+                    NewExtDlg_OnOK(hwndDlg, s_pNewExt);
+                    break;
+                case IDCANCEL:
+                    EndDialog(hwndDlg, IDCANCEL);
+                    break;
+                case IDC_NEWEXT_ADVANCED:
+                    s_pNewExt->bAdvanced = !s_pNewExt->bAdvanced;
+                    NewExtDlg_OnAdvanced(hwndDlg, s_pNewExt);
+                    break;
+            }
+            break;
+    }
+    return 0;
+}
+
+static BOOL
+FileTypesDlg_AddExt(HWND hwndDlg, LPCWSTR pszExt, LPCWSTR pszFileType)
+{
+    DWORD dwValue = 1;
+    HKEY hKey;
+    WCHAR szKey[MAX_PATH];
+    LONG nResult;
+
+    // Search the next "ft%06u" key name
```

Comment:
So this is also what Windows does, i.e. it uses by default a "ft..." keyname? Also, only 6 digits?

Reply:
![decimal](https://user-images.githubusercontent.com/2107452/40204231-bd192382-5a62-11e8-8205-c9e968669922.png)
Digits are decimal. And max 13 characters ("ft4294967295" + "\0"). OK.